tests/framework/utils.py

```diff
@@ -611,14 +611,16 @@ def compare_versions(first, second):
     return 0
 
 
-def get_kernel_version():
+def get_kernel_version(include_patch=True):
     """Return the current kernel version in format `major.minor.patch`."""
     linux_version = platform.release()
     for idx, char in enumerate(linux_version):
         if not char.isdigit() and char != '.':
             linux_version = linux_version[0:idx]
+            if not include_patch:
+                major, minor, *_rest = linux_version.split(".")
+                linux_version = f"{major}.{minor}"
```

Comment:
I understand you suggest to have default level 2. That means I need to change all the code that used it before this PR which used level 3. This would make this PR even more cumbersome.

Reply:
I thought level 2 == patch. Because The levels start from 0.

Reply:
Sorry, actually to deduplicate a bit I think we can do something like:

```
        if char == '.':
            actual_level += 1
        if actual_level > level or (not char.isdigit() and char != '.')
            linux_version = linux_version[0:idx]
            break
```